google-cloud-examples/src/main/java/com/google/cloud/examples/storage/snippets/StorageSnippets.java

```diff
@@ -1577,4 +1582,141 @@ public URL generateV4GPutbjectSignedUrl(String bucketName, String objectName)
     // [END storage_generate_upload_signed_url_v4]
     return url;
   }
+
+  /** Example of creating an HMAC key for a service account * */
+  public HmacKey createHmacKey(String serviceAccountEmail) throws StorageException {
+    // [START storage_create_hmac_key]
+    // Instantiate a Google Cloud Storage client
+    Storage storage = StorageOptions.getDefaultInstance().getService();
+
+    // The service account email for which the new HMAC key will be created.
+    // String serviceAccountEmail = "service-account@iam.gserviceaccount.com";
+    ServiceAccount account = ServiceAccount.of(serviceAccountEmail);
+    HmacKey hmacKey = storage.createHmacKey(account);
+
+    String secret = hmacKey.getSecretKey();
+    HmacKeyMetadata metadata = hmacKey.getMetadata();
+
+    System.out.println("The Base64 encoded secret is: " + secret);
+    System.out.println("Do not miss that secret, there is no API to recover it.");
+    System.out.println("The HMAC key metadata is:");
+    System.out.println("ID: " + metadata.getId());
+    System.out.println("Access ID: " + metadata.getAccessId());
+    System.out.println("Project ID: " + metadata.getProjectId());
+    System.out.println("Service Account Email: " + metadata.getServiceAccount().getEmail());
+    System.out.println("State: " + metadata.getState().toString());
+    System.out.println("Time Created: " + new Date(metadata.getCreateTime()).toString());
+    System.out.println("Time Updated: " + new Date(metadata.getUpdateTime()).toString());
+    System.out.println("ETag: " + metadata.getEtag());
+    // [END storage_create_hmac_key]
+    return hmacKey;
+  }
+
+  /** Example of retrieving the metadata of an existing HMAC key. * */
+  public HmacKeyMetadata getHmacKey(String accessId) throws StorageException {
```

Comment:
pass in projectId in this example and the others as well. It's a small level of detail but it keeps consistency.